mirrord/sip/src/lib.rs

```diff
@@ -341,13 +341,20 @@ mod main {
         if whitespace::skip(rest).starts_with('[') {
             None
         } else {
-            content
+            let mut iterator = content
                 .split_once('\n')
                 .map(|(line, _)| line)
                 .unwrap_or(content)
-                .split_whitespace()
-                .next()
-                .map(ToString::to_string)
+                .split_whitespace();
+
+            let first_token = iterator.next()?;
+            // if first_token is just #!, keep going
+            if first_token == "#!" {
+                let executable = iterator.skip_while(|s| s.is_empty()).next()?;
+                Some(format!("{first_token}{executable}"))
```

Comment:
Looks like we use the length of the result of this function to find the start fo the rest of the file in `patch_script`, so this could cause bugs.

Example:
first line is `#!<7 spaces>/bin/sh node`
then the new patched script would be
`#!<mirrord-tmp-dir>/bin/sh/bin/sh node`

So if we don't want to make a bigger change to the sip code, this function should return a string that is equal to the shebang as it is in the file, including whitespace.

Reply:
I don't see it this way - reserving the behavior doesn't have any implication other than making our code more confusing..